docs/sources/developers/plugins/add-authentication-for-data-source-plugins.md

```diff
@@ -289,58 +289,71 @@ If your data source uses the same OAuth provider as Grafana itself, for example
 
-When configured, Grafana will pass the user's token to the plugin in an Authorization header, available on the `QueryDataRequest` object on the `QueryData` request in your backend data source.
+When configured, Grafana can forward authorization HTTP headers, like `Authorization` or `X-ID-Token`, to an external server. For doing so, it's just necessary to create the HTTP client using the [Grafana Plugin SDK](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go). This also ensures that the request has some recommended settings, like a default timeout.
 
 ```go
-func (ds *dataSource) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
-	token := strings.Fields(req.GetHTTPHeader(backend.OAuthIdentityTokenHeaderName))
-	var (
-		tokenType   = token[0]
-		accessToken = token[1]
-	)
-
-	// ...
-	return &backend.CheckHealthResult{Status: backend.HealthStatusOk}, nil
+func NewDatasource(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	opts, err := settings.HTTPClientOptions()
+	if err != nil {
+		return nil, fmt.Errorf("http client options: %w", err)
+	}
+  // Important to reuse the same client for each query, to avoid using all available connections on a host
+	cl, err := httpclient.New(opts)
+	if err != nil {
+		return nil, fmt.Errorf("httpclient new: %w", err)
+	}
+	return &Datasource{
+		httpClient: cl,
+	}, nil
 }
 
 func (ds *dataSource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
-  token := strings.Fields(req.GetHTTPHeader(backend.OAuthIdentityTokenHeaderName))
-	var (
-		tokenType   = token[0]
-		accessToken = token[1]
-	)
-
-	for _, q := range req.Queries {
-		// ...
-	}
+    // Important to keep the Context, since the injected middleware is configured there
+    req, err := http.NewRequestWithContext(ctx, http.MethodGet, "https://some-url", nil)
+    if err != nil {
+      return nil, fmt.Errorf("new request with context: %w", err)
+    }
+    // Authorization header will be automatically injected if oauthPassThru is configured
+    resp, err := ds.httpClient.Do(req)
+    // ...
 }
 ```
 
-In addition, if the user's token includes an ID token, Grafana will pass the user's ID token to the plugin in an `X-ID-Token` header, available on the `QueryDataRequest` object on the `QueryData` request in your backend data source.
+You can see a full working example here: [datasource-http-backend](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/datasource-http-backend).
+
+The same approach works with `CheckHealth` and `CallResource` requests.
```

Comment:
Can probably remove this line if we mention which requests are supported up front above!

```suggestion

```